firewood/src/shale/mod.rs

```diff
@@ -134,22 +134,22 @@ impl<T: Storable> Obj<T> {
     }
 
     pub fn flush_dirty(&mut self) {
-        if !self.value.is_mem_mapped() {
-            if let Some(new_value_len) = self.dirty.take() {
-                let mut new_value = vec![0; new_value_len as usize];
-                // TODO: log error
-                self.value.write_mem_image(&mut new_value).unwrap();
-                let offset = self.value.get_offset();
-                let bx: &mut dyn CachedStore = self.value.get_mut_mem_store();
-                bx.write(offset, &new_value);
-            }
+        if let Some(new_value_len) = self.dirty.take() {
```

Comment:
This might seem like a nit, but I actually don't think it is... your change is an optimization for calling `flush_dirty` and should therefore exist in the `flush_dirty` code instead.

```suggestion
        // faster than calling `self.dirty.take()` on a `None`
        if self.dirty.is_none() {
            return;
        }

        if let Some(new_value_len) = self.dirty.take() {
```